test/perf/performance-test-cases.js

```diff
@@ -0,0 +1,21 @@
+const examples = [
+  {
+    name: 'instancing',
+    animationLoop: require('../../examples/core/instancing/app').default,
+    targetFPS: 40
+  }
+];
+
+export default examples.map(({name, animationLoop, targetFPS}) => {
+  return {
+    name,
+    onInitialize: animationLoop.onInitialize.bind(animationLoop),
```

Comment:
- Correct, `TestRunner` manages a shared `AnimationLoop` and expose its animation props to each test case. This is so that we don't end up creating one canvas per test (aside from competing for resources, they also interfere with on-screen logging).
- `TestRunner` accepts the following test case format:

    ```js
    testRunner.add({
      name: 'test-case-1',
      animationLoop: new MyAnimationLoop(),
      goldenImage: './test-case-1.png'
    });
    ```

  However in this particular case we need to manually overwrite some animation props before calling `onRender`, hence the binding.

- I did this to minimize the change to the example app. I personally find it hard to maintain when the example needs to deal with a lot of implicit use cases.

Reply:
Agreed we should not change the example apps unless necessary. I was mainly exploring if some of the stuff here could be generalized.

> This is so that we don't end up creating one canvas per test (aside from competing for resources, they also interfere with on-screen logging).

I'm totally fine with this PR as is, already approved, just brainstorming: 

Doesn't `AnimationLoop` accept a canvas/context? I seem to recall canvas creation being intended as a convenience for writing compact examples. Should not be too hard to make e.g. `start` override an existing loop's context?

Reply:
Hmm that could work. I'll give it a try.